modules/serverless-apiserversource.adoc

```diff
@@ -0,0 +1,120 @@
+
+.Example ApiServerSource YAML
+
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1alpha1
+      kind: Broker
+      name: default
+----
+
+== Prerequisites
+* Using the ApiServerSource requires you to have Knative Serving and Eventing installed.
+// For more information, see the documentation on xref:../.../serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[Installing OpenShift Serverless].
+
+== Procedure
+// . Create the `default` broker in your namespace. For more information, see the documentation on xref:../.../serverless/knative_eventing/serverless-broker.adoc[Creating the broker].
+. Create a Service Account that the `ApiServerSource` runs as. Create a file named `serviceaccount.yaml` and copy the sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default
+----
++
+If you want to re-use an existing Service Account with the appropriate permissions, you need to modify the `serviceaccount.yaml`.
+
+To create the service account from `serviceaccount.yaml`:
++
+----
+$ oc apply --filename serviceaccount.yaml
+----
+
+. Create a file named `k8s-events.yaml` and copy sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1alpha1
+      kind: Broker
+      name: default
+----
++
+[NOTE]
+====
+If you want to consume events from a different namespace or use a different Service Account, you need to modify `k8s-events.yaml` accordingly.
+====
++
+Create the event source:
++
+----
+$ oc apply --filename k8s-events.yaml
+----
+
+== Verfiy the ApiServerSource is working using a trigger
+
+To check that the `ApiServerSource` is set up correctly, create a Knative service that dumps incoming messages to its log, and creates a trigger from the broker to that Knative service.
+
+// add links to trigger docs
```

Comment:
yep, if we want to show the broker as the sync for the ApiServerSource, than we need a trigger, which will lift the events from the broker to the KSVC (Knative Serving Service), apply the filter rules.

Reply:
Oh, I'd suggest showing two filters, where both filter on different event types, and than routing the events to different KSVCs.

Here is a list of event types, that the `ApiServerSource` emits:

https://github.com/knative/eventing/blob/master/config/core/resources/apiserversource.yaml#L28-L33

Reply:
Just waiting for QE to approve my trigger PR so I have something to link to. No point rewriting it in another PR :woman_shrugging: